dpdk/vr_dpdk_interface.c

```diff
@@ -361,13 +447,6 @@ dpdk_agent_if_add(struct vr_interface *vif)
     RTE_LOG(INFO, VROUTER, "Adding vif %u packet device %s\n",
                 vif->vif_idx, vif->vif_name);
 
-    /* check if packet device is already added */
-    if (vr_dpdk.packet_ring != NULL) {
-        RTE_LOG(ERR, VROUTER, "\terror adding packet device %s: already exist\n",
-                vif->vif_name);
-        return -EEXIST;
-    }
-
```

Comment:
There is a check in dp-core vrouter_add_interface() if the vif is already exist. But you are right, one could can add two agent interfaces with different vif_idx, so I put the check back.